docs/src/main/asciidoc/_elytron/Elytron_WildFly_Java_Security_Manager.adoc

```diff
@@ -0,0 +1,326 @@
+
+[[Elytron_JSM_Further_Background]]
+== Further background
+Additionally to the documentation of https://jboss-modules.github.io/jboss-modules/manual/#security-manager[JBoss Modules and the Security Manager] the following explanations have been extracted from an WildFly developer conversation.
+
+[[Elytron_JSM_Protection_Domains]]
+=== Elytron WildFly Security Manager and Protection Domains
+Within WildFly there are two types of ProtectionDomain:
+
+* Server module. (The stuff under the modules folder).
+* Deployment.
+
+Server modules automatically have the {oracle-javadoc}/java.base/java/security/AllPermission.html[AllPermission] granted.
+Deployments have a combination of their permission.xml as well as possibly the minimal permission set from the security manager subsystem.
+
+==== Permission checks
+As a call progresses from class to class, module to module, deployment to module a list of all of the protection domains of each of these builds up.
+When a permission check is performed it will only success if each and every protection domain on the call stack has been granted the permission.
+
+This is where doPrivileged comes in, this is effectively saying "I have verified this is safe, forget the protection domains that called me."
+So when a deployment calls into a server module and that server module calls doPrivileged then the deployments module will not longer be a part of the permission check.
+
+Adding a doPrivileged seems an obvious solution but when added we also need to think about how deployments could abuse this to get their protection domain dropped from the permissions check.
+
+==== Privileged blocks in dependencies
+Adding the privileged block in a dependency used by WildFly works because the protection domain of the server modules has all the permissions I mean, is it because the current protection domain where the dependency is has all permissions granted, or is it because it is running on a privileged block?
+
+One point to start with there is nothing special about a PrivilegedAction - it is just a special interface which allows some form of Runnable to be passed in which also has a return type. If this was added later they could have just used the functional interfaces. The special part is the doPrivileged call.
+Under normal circumstances every jar would be represented by it's own protection domain and each jar could be assigned it's own permissions. When the security manager performs it's permissions check it would make sure every protection domain in the call stack has been granted that permission.
+
+Our case is slightly special as all of our modules just get granted AllPermission, so we end up in a situation where deployments have a defined set of permissions and our modules have all - by default the security manager checks the permissions of both of these during a permission check that spans them - our module will of course always pass.
```

Comment:
@boris-unckel  I know this is taken from zulip. But when reading this documentation it is not clear what "our case" mean

Reply:
```The WildFly modules case is``` ...